docs/source/tutorial/test_a_project.md

```diff
@@ -0,0 +1,335 @@
+
+Let us explore what this looks like in practice. Consider the node function `split_data` defined in the data science pipeline:
+
+<details>
+<summary><b>Click to expand</b></summary>
+
+```python
+def split_data(data: pd.DataFrame, parameters: Dict) -> Tuple:
+    """Splits data into features and targets training and test sets.
+
+    Args:
+        data: Data containing features and target.
+        parameters: Parameters defined in parameters_data_science.yml.
+    Returns:
+        Split data.
+    """
+    X = data[parameters["features"]]
+    y = data["price"]
+    X_train, X_test, y_train, y_test = train_test_split(
+        X, y, test_size=parameters["test_size"], random_state=parameters["random_state"]
+    )
+    return X_train, X_test, y_train, y_test
+```
+
+</details>
+
+The function takes a pandas DataFrame and dictionary of parameters as input, and splits the input data into 4 different data objects as per the parameters provided. We recommend following [pytest's anatomy of a test](https://docs.pytest.org/en/7.1.x/explanation/anatomy.html#anatomy-of-a-test) which breaks a test down into 4  steps: arrange, act, assert, and cleanup. For this specific function, these steps will be:
+
+1. Arrange: Prepare the inputs `data` and `parameters`.
+2. Act: Make a call to `split_data` and capture the outputs with `X_train`, `X_test`, `Y_train`, and `Y_test`.
+3. Assert: Ensure that the length of the outputs are the same as the expected lengths
+
+Cleanup is not required for this test. Remember to import the function being tested and any necessary modules at the top of the file.
+
+When we put these steps together, we have the following test:
+
+<details>
+<summary><b>Click to expand</b></summary>
+
+```python
+# NOTE: This example test is yet to be refactored.
+# A complete version is available under the testing best practices section.
+
+import pandas as pd
+from spaceflights.pipelines.data_science.nodes import split_data
+
+    def test_split_data():
+        # Arrange
+        dummy_data = pd.DataFrame(
+        {"engines": [1, 2, 3],
+         "crew": [4, 5, 6],
+         "passenger_capacity": [5, 6, 7],
+         "price": [120, 290, 30]})
+
+        dummy_parameters = {"model_options":
+                     {"test_size": 0.2,
+                      "random_state": 3,
+                      "features": ["engines", "passenger_capacity", "crew"]}}
+
+        # Act
+        X_train, X_test, y_train, y_test = split_data(dummy_data, dummy_parameters["model_options"])
+
+        # Assert
+        assert len(X_train) == 2
+        assert len(y_train) == 2
+        assert len(X_test) == 1
+        assert len(y_test) == 1
+```
```

Comment:
Let's have a unhappy case and some failure cases too. @Nab-0 has a good tutorial on this

Reply:
Added an unhappy/negative test case in [91e9ec0](https://github.com/kedro-org/kedro/pull/3782/commits/91e9ec07627f8e45082b8b5143510a64c72df645). 

I'm wary about adding errors to `split_data` to be found by failing cases as users may stop the tutorial before going through testing, incorrectly assuming that the implemented logic is sound. 

I considered adding a section about Test Driven Development instead, but it's difficult to recommend TDD in the tutorial, as the tutorial itself does not practice TDD 😅. Any thoughts?

Reply:
I think explaining TDD here goes a bit beyond just best practices for coding, because it goes into actual ways of working which is a whole other story and not really something I feel like we have a firm stance on as the Kedro team.